src/components/ChallengeModal/ChallengeModal.test.tsx

```diff
@@ -0,0 +1,543 @@
+/*
+ * Copyright (C)  Online-Go.com
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import * as React from "react";
+import { render, screen, fireEvent } from "@testing-library/react";
+import { ChallengeModalBody, ChallengeModalProperties } from "./ChallengeModal";
+import { JGOFTimeControl, JGOFTimeControlSpeed, JGOFTimeControlSystem } from "goban";
+import { RuleSet } from "@/lib/types";
+import { post } from "@/lib/requests";
+
+type ChallengeModes = "open" | "computer" | "player" | "demo";
```

Comment:
I was thinking export the original type, but I realize this type isn't necessary at all if you use `as const` below.